gfw_pixetl/data_type.py

```diff
@@ -47,15 +49,19 @@ def has_no_data(self):
 
     @staticmethod
     def _validate_no_data(
-        data_type: str, no_data: Optional[Union[int, float]], nbits: Optional[int]
+        data_type: str,
+        no_data: Optional[Union[StrictInt, float]],
+        nbits: Optional[StrictInt],
     ):
         dtype = data_type.lower()
 
         if "int" in dtype and (no_data is not None and not isinstance(no_data, int)):
             message = f"No data value {no_data} must be of type `int` or None for data type {data_type}"
             raise ValueError(message)
-        elif ("float" in dtype or dtype in ["half", "single", "double"]) and (
-            no_data is not None and not math.isnan(no_data)
+        elif (
+            ("float" in dtype or dtype in ["half", "single", "double"])
```

Comment:
nitpick: I think those list of dtypes might have constants defined for them in numpy you can use

Reply:
I came-up with this. It's not easier to read, but maybe it's better?
`getattr(DataTypeEnum, dtype, None) is not None and np.issubdtype(np.dtype(dtype), np.floating)`
What do you think?